File: javascript/pie.js
function Pie(filling) {
  this.fill_with(filling);
  this.eaten = false;
};

Pie.prototype.eat = function() {
  this.eaten = true;
};

Pie.prototype.fill_with = function(filling) {
    this.filling = filling;
};

var apple_pie = new Pie('apple');
apple_pie.eat

var frame = '';
for (i = 0; i < 28; i++) {
    frame = frame + '♦'
}

console.log(frame)
console.log("♦ Who ate the " + apple_pie.filling + " pie?!? ♦")
console.log(frame)

File: ruby/pie.rb
require 'opal'

class Pie

  attr_reader :filling

  def initialize (filling)
    fill_with(filling)
    @eaten = false
  end

  def eat
    @eaten = true
  end

  private

    def fill_with (filling)
      @filling = filling
    end

end

apple_pie = Pie.new(:apple)
apple_pie.eat

frame = (0..27).reduce('')  { |sum, n| sum + '♦' }

puts frame
puts "♦ Who ate the #{apple_pie.filling} pie?!? ♦"
puts frame
